Jenkinsfile

```diff
@@ -520,16 +522,48 @@ def cleanDir(path){
 }
 
 def dotnet(Closure body){
-  def dockerTagName = 'docker.elastic.co/observability-ci/apm-agent-dotnet-sdk-linux:latest'
-  sh label: 'Docker build', script: "docker build --tag ${dockerTagName} .ci/docker/sdk-linux"
+
   def homePath = "${env.WORKSPACE}/${env.BASE_DIR}"
-  docker.image("${dockerTagName}").inside("-e HOME='${homePath}' -v /var/run/docker.sock:/var/run/docker.sock"){
+  withEnv([
+    "HOME=${homePath}",
+    "DOTNET_ROOT=${homePath}/.dotnet",
+    "PATH+DOTNET=${homePath}/.dotnet/tools:${homePath}/.dotnet"
+    ]){
+    sh(label: 'Install dotnet SDK', script: """
+    mkdir -p \${DOTNET_ROOT}
+    # Download .Net SDK installer script
+    curl -s -O -L https://dotnet.microsoft.com/download/dotnet-core/scripts/v1/dotnet-install.sh
+    chmod ugo+rx dotnet-install.sh
+
+    # Install .Net SDKs
+    ./dotnet-install.sh --install-dir "\${DOTNET_ROOT}" -version '2.1.505'
+    ./dotnet-install.sh --install-dir "\${DOTNET_ROOT}" -version '3.0.103'
+    ./dotnet-install.sh --install-dir "\${DOTNET_ROOT}" -version '3.1.100'
+    ./dotnet-install.sh --install-dir "\${DOTNET_ROOT}" -version '5.0.203'
+    """)
```

Comment:
Better to move it to the https://github.com/elastic/apm-agent-dotnet/tree/master/.ci/linux folder and then add some retries in case something bad happened

Reply:
I will move it to an step to the library